app/grandchallenge/algorithms/models.py

```diff
@@ -156,10 +157,23 @@ class Algorithm(UUIDModel, TitleSlugDescriptionModel):
     )
     use_flexible_inputs = models.BooleanField(default=True)
     repo_name = models.CharField(blank=True, max_length=512)
+    image_requires_gpu = models.BooleanField(default=False)
+    image_requires_gpu_memory_gb = models.PositiveIntegerField(default=4)
+    image_requires_memory_gb = models.PositiveIntegerField(default=4)
+    image_requires_cpu_cores = models.DecimalField(
+        default=Decimal("1.0"), max_digits=4, decimal_places=2
+    )
```

Comment:
`image_requires_gpu_memory_gb` and `image_requires_cpu_cores` are never used on the image and do not need to be set, so can be removed. We can add them if we ever need it. No need to remove the fields on the `ComponentImage` for now.

Would be nice if `image_requires_gpu` and `image_requires_memory_gb` also provide the initial values to the form when uploading a new container via the web UI.